### PR TITLE
[Ready for review] Graph & Key Vault MGMT

### PR DESCRIPTION
**Pull request by jianghaolu**

### CODE REVIEW

**Comment by jianghaolu** [comment]
@martinsawicki 
This will show a diff without generated files: https://github.com/Azure/azure-sdk-for-java/compare/0484ce6...93be0f4

**Comment by martinsawicki** [comment]
hmm, that view doesn't let me add comments..

**Comment by martinsawicki** [comment]
In general, from my quick look, the following 3 key things caught my eye:
- What is called "Group" should probably better be called "ActiveDirectoryGroup" (based on the naming of ADGroupInner)
- Replace Boolean returning methods with boolean
- Javadoc's @return should start with a small letter and no period
After the PR, i'll try to play with in in code to see if I'd have more comments. 
I noticed some interesting uses of base interfaces to reuse teh same definition in different contexts - i'll play with that and see if we can reuse taht pattern in other places (does it really avoid raw type refs? It';d be great if it did, without the duplication)

File: azure-mgmt-keyvault/src/main/java/com/microsoft/azure/management/keyvault/Vault.java [hunk 1]
```diff
@@ -25,12 +25,12 @@
         Updatable<Vault.Update>,
         Wrapper<VaultInner> {
     /**
-     * @return The URI of the vault for performing operations on keys and secrets.
+     * @return the URI of the vault for performing operations on keys and secrets.
```
Comment (martinsawicki) [inline azure-mgmt-keyvault/src/main/java/com/microsoft/azure/management/keyvault/Vault.java]: also, no period at the end for @return :-)

**Comment by jianghaolu** [comment]
Thanks! Good catches. I'll fix them.

Two things to notice:
1) The `Azure.authenticate()` signature is changed
2) Graph is in fact in progress. E.g.: `Group` doesn't have an `impl` yet. I'll add them in a separate PR.

**Pull request merged**